client/app/components/QueryBasedParameterInput.jsx

```diff
@@ -41,19 +42,18 @@ export class QueryBasedParameterInput extends React.Component {
     }
   }
 
-  _loadOptions(queryId) {
+  async _loadOptions(queryId) {
     if (queryId && (queryId !== this.state.queryId)) {
       this.setState({ loading: true });
-      Query.dropdownOptions({ id: queryId }, (options) => {
-        if (this.props.queryId === queryId) {
-          this.setState({ options, loading: false });
+      const options = await this.props.parameter.loadDropdownValues();
+      if (this.props.queryId === queryId) {
```

Comment:
I think just adding a comment `// stale queryId check` would make it clear for both patterns.

Reply:
If that's the case, I'd favor `await` over Promises. Not sure why we haven't begun embracing `await`s elsewhere in the code?

Reply:
> Not sure why we haven't begun embracing awaits elsewhere in the code?

We try to leave things for future refactors 😎 

> but with await this fragment looks linear, and this condition looks confusing

But `await` itself signals that this is async operation. Maybe it's just confusing because we're less used to using it?

Let's give it a try and see how goes.

Reply:
_Do we_ care if the transpiled version is using async/await natively? Feels meaningless to me but I may be missing something?

Reply:
Took care of it https://github.com/getredash/redash/pull/3609